### DISCUSSION TITLE
Python API - None object returned

### DISCUSSION THREAD

**Discussion by gtidei** [chipsalliance/UHDM#946] [Q&A]
After generating the surelog.uhdm file I have tried to run the example shown on that

import uhdm

s = uhdm.Serializer()
data = s.Restore('surelog.uhdm')
module_iterator = uhdm.vpi_iterate(uhdm.uhdmallModules,data[0])

while(True):
   vpiObj_module = uhdm.vpi_scan(module_iterator)
   if vpiObj_module is None:
       break
   #print the string pointed by vpiName attribut of vpiObj_module
   print(uhdm.vpi_get_str(uhdm.vpiName,vpiObj_module))
   
   
However a 'None' string is printed. The uhdm file seems to be correctly generated, I have also tried another surelog.uhdm file not generated by me by present here in Github. I might therefore assume that there's something wrong in the way I have installed the Python API (even tough though at least functions are called without problems).

![image](https://github.com/chipsalliance/UHDM/assets/130081328/d3b3e8af-24ba-4d99-b422-fcb685926beb)

After the build I have placed the _py_uhdm.so and uhdm.py files into the python3 distribution packages
![image](https://github.com/chipsalliance/UHDM/assets/130081328/c6fdd7cc-ec50-4fa5-b63d-cb54d41a12e3)
import uhdm works correctly but there's something I am missing

**Comment by Thomasb81** [comment]
> I have also tried another surelog.uhdm file not generated by me by present here in Github.

If your uhdm file, is not generated with the same  version of uhdm model compile in the python module, it will not work.
So there is very few chance to work if you do not fully know what source code has been use to generate the uhdm file you found.

 You can try to reproduce the unitest that use build-in data :  https://github.com/chipsalliance/UHDM/blob/master/python/tests/test_module.py

You can also re-run the non-regression :https://github.com/chipsalliance/UHDM/blob/9a722544b358204908bd950cd276945eb01897b1/.github/workflows/main.yml#L94
